python/plugins/processing/script/ScriptTemplate.py

```diff
@@ -85,15 +80,25 @@ def groupId(self):
         """
         return "examplescripts"
 
-    def shortHelpString(self):
+    def shortHelpString(self) -> str:
         """
         Returns a localised short helper string for the algorithm. This string
         should provide a basic description about what the algorithm does and the
-        parameters and outputs associated with it..
+        parameters and outputs associated with it.
+        """
+        return "Example algorithm short description"
+
+    def icon(self) -> QIcon:
+        """
+        Returns the icon to use in the Processing toolbox.
         """
-        return self.tr("Example algorithm short description")
+        # If this method is removed, the default Processing icon will be used.
+        # As this example is doing a copy of the input to the output,
+        # we use the "duplicate" layer icon.
+        # You can use "PyQGIS Resource Browser" plugin to find icons
+        return QIcon(":/images/themes/default/mActionDuplicateLayer.svg")
```

Comment:
I think this format is a bit confusing, and in general using resources for plugins is discouraged in modern PyQt development. (It also is much trickier for PyQt6).

It would be better to use a hardcoded file path instead, as that is immediately understandable for readers, but I realise that it's tricky here! We could get away with a hardcoded invalid path, since that will return an invalid QIcon but not raise any exceptions...

Reply:
What about `QgsApplication.getThemeIcon('mActionDuplicateLayer.svg')` ? Is-it also discouraged ?

For a Python script, it looks indeed tricky to use a hard-coded file path.

What about adding a snippet using a base64 string ? (we can comment this example for instance) :

```
    def icon(self) -> QIcon:
        data = "Your base64 encoded string"
        pixmap = QPixmap()
        pixmap.loadFromData(base64.b64decode(data))
        return QIcon(pixmap)
```

Reply:
> What about QgsApplication.getThemeIcon('mActionDuplicateLayer.svg') ? Is-it also discouraged ?

Not discouraged, but also not something I'd think is particularly useful for a user-created algorithm (since you're restricted to the choice of inbuilt icons)

>     def icon(self) -> QIcon:
        data = "Your base64 encoded string"
        pixmap = QPixmap()
        pixmap.loadFromData(base64.b64decode(data))
        return QIcon(pixmap)

If you've got a dummy string there anyway, it's much simpler to just put a dummy hardcoded path there and let the developer work out the logic needed to distribute this icon.

Maybe it's better to just omit icon() in order to keep this template as simple as possible..

Reply:
Ok, I have removed it ;-)